src/__tests__/models/Logger.test.ts

```
@@ -1,519 +1,296 @@
+  });
 
-    consoleLogSpy.mockRestore();
+  it("respects NO_COLOR env by disabling ANSI color codes", async () => {
+    const prev = process.env.NO_COLOR;
+    process.env.NO_COLOR = "1";
+    try {
+      const logger = createLogger({ threshold: "info" });
+      await logger.info("plain");
+      const outputs = gather();
+      // No ESC sequences should be present
+      expect(outputs).not.toMatch(/\x1b\[/);
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)